### PR TITLE
Moved submission mobx stuff into its own class

### PR DESCRIPTION
**Pull request by pureooze**
This will allow us to call a setter in UI tests to setup the store for test scenarios without having to stub Hypermedia APIs.

### CODE REVIEW

File: components/d2l-activity-editor/d2l-activity-assignment-editor/state/assignment.js [hunk 1]
```diff
@@ -87,11 +94,8 @@ export class Assignment {
...
 		this.isOriginalityCheckEnabled = entity.isOriginalityCheckEnabled();
 		this.isGradeMarkEnabled = entity.isGradeMarkEnabled();
-		this.submissionTypeOptions = entity.submissionTypeOptions();
 		this.allCompletionTypeOptions = entity.allCompletionTypeOptions();
-		this.canEditSubmissionType = entity.canEditSubmissionType();
 		this.canEditCompletionType = entity.canEditCompletionType();
-		this.submissionType = String(entity.submissionType().value);
 		this.completionType = entity.completionTypeValue();
 
 		this.canEditSubmissionsRule = entity.canEditSubmissionsRule();
```
Comment (pureooze) [inline components/d2l-activity-editor/d2l-activity-assignment-editor/state/assignment.js:101]: Not sure if submission rules should be part of this new class too? Thoughts?

**Comment by pureooze** [comment]
Uhh didnt see any replies so I deleted my comment because I was pushing an update. Yay eventual consistency?...

File: components/d2l-activity-editor/d2l-activity-assignment-editor/state/assignment.js [hunk 2]
```diff
@@ -180,6 +184,10 @@ export class Assignment {
 		this.instructions = value;
 	}
 
+	setAssignmentSubmissionType(assignmentSubmissionType) {
```
[the remark this answers is not in the document]
Comment (pureooze) [reply components/d2l-activity-editor/d2l-activity-assignment-editor/state/assignment.js:186]: `setSubmissionType` is supposed to be for updating the selected `<option>` in the submission type dropdown where as `setAssignmentSubmissionType` was for using in tests to overwrite the whole `AssignmentSubmissionType` object. I think my naming choice for the class probably causes some confusion here so I'll try to come up with a better name. 🤔

File: components/d2l-activity-editor/d2l-activity-assignment-editor/d2l-activity-assignment-editor-submission-and-completion.js [hunk 3]
```diff
@@ -151,19 +151,22 @@ class ActivityAssignmentSubmissionAndCompletionEditor extends ActivityEditorMixi
...
 				</label>
 			</div>
 		`;
 	}
 	_renderAssignmentSubmissionsRule(assignment) {
-		if (!assignment || !assignment.showSubmissionsRule) {
+
+		if (!assignment ||
+			!assignment.assignmentSubmissionProps ||
+			!assignment.assignmentSubmissionProps.showSubmissionsRule) {
```
Comment (pureooze) [inline components/d2l-activity-editor/d2l-activity-assignment-editor/d2l-activity-assignment-editor-submission-and-completion.js:165]: Unfortunately cant use optional chaining here, `polymer test` does not seem to like `assignment?.assignmentSubmissionProps?.showSubmissionsRule`

File: components/d2l-activity-editor/d2l-activity-assignment-editor/d2l-activity-assignment-editor-submission-and-completion.js [hunk 4]
```diff
@@ -131,19 +135,21 @@ class ActivityAssignmentSubmissionAndCompletionEditor extends ActivityEditorFeat
 	_setSubmisisonsRule(e) {
 		const assignment = store.getAssignment(this.href);
 		const data = e.target.value;
-		assignment && assignment.setSubmissionsRule(data);
+		assignment && assignment.assignmentSubmissionProps.setSubmissionsRule(data);
```
Comment (kieranderson) [inline components/d2l-activity-editor/d2l-activity-assignment-editor/d2l-activity-assignment-editor-submission-and-completion.js:138]: Possible for assignment to be null here?

**Pull request merged**